log/handler.go

```diff
@@ -81,6 +85,34 @@ func NetHandler(network, addr string, fmtr Format) (Handler, error) {
 	return closingHandler{conn, StreamHandler(conn, fmtr)}, nil
 }
 
+// SplunkHandler posts data to Splunk HTTP Event Collector.
+func SplunkHandler(url, token, source, sourcetype, index string, skipTLSVerify bool, fmtr Format,
+	originalHandler Handler) (Handler, error) {
+	originalLogger := logger{[]interface{}{}, new(swapHandler)}
+	originalLogger.SetHandler(originalHandler)
+	tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: skipTLSVerify}}
+	httpClient := &http.Client{Timeout: time.Second * 20, Transport: tr}
+	splunkClient := splunk.NewClient(
+		httpClient,
+		url,
+		token,
+		source,
+		sourcetype,
+		index,
+	)
+	writer := &splunk.Writer{
+		Client:     splunkClient,
+		MaxRetries: 2,
+	}
+	go func() {
+		for {
+			err := <-writer.Errors()
+			originalLogger.Warn("Error sending info to Splunk", "err", err)
+		}
+	}()
+	return StreamHandler(writer, fmtr), nil
```

Comment:
Actually, now that I am taking a closer look at this function, I don't fully understand why you chose to even use an `io.Writer` for logging to splunk via HTTP. If you implement the handler in the splunk package, maybe consider simply implementing the `log.Handler` interface in the `splunk.Client`, such that the `Log` method sends a single event.

Reply:
Thanks for the review Felix. I followed your comments to simplify the code.